### PR TITLE
adds upgrade guide draft (frame->span + stream open)

### PR DESCRIPTION
**Pull request by dvdsk**
We will point users who upgrade to the next rodio release to this document. It will help them upgrade (and save us a lot of confused users and issues).

I do not think I have missed anything. I suggest we merge this and then every PR that introduces a breaking change updates this document. I have already done that for still open PR #663.

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This guide will help you update your code when upgrading from rodio 0.20 or earlier to the current version.
+
+## Source implementations
+- Source had a required method `current_frame_len`. In the latest version of rodio *frame* has been renamed to *span*. You will need to change every occurrence of `current_frame_len` to `current_span_len`.
+
+## OutputStream
+- The outputstream is now more configurable. Where you used `OutputStream::try_default()` you have a choice:
+    - *(recommended)* Get an error when the default stream could not be opened: `OutputStreamBuilder::open_default_stream()?`
+    - Keep the old behavior using: `OutputStreamBuilder::open_stream_or_fallback()`, which tries to open the default (audio) stream trying others when the default stream could not be opened.
```
Comment (PetrGlad) [inline UPGRADE.md:9]: I am always forgetting to mention this, but there is also a change in the fallback logic. Previous version either probed default settings of all devices or all settings of the default device. Current implementation tries all the combinations. I hope that does not break anything.

Comment (dvdsk) [reply UPGRADE.md:9]: I'll update the guide :+1:

Comment (dvdsk) [reply UPGRADE.md]: take a look, if this is not right please let me know what you would write instead.

**Comment by PetrGlad** [comment]
This will not be the last update that needs some review or changes for users.
I would add a top level heading that states the versions like "Upgrade from versions before 0.20.1 to 0.20.1"

**Comment by dvdsk** [comment]
done

**Review by PetrGlad — APPROVED** [review]

**Pull request merged**